testing/README.md

```diff
@@ -0,0 +1,355 @@
+# `import {Testing} from 'Shopify'`
+
+This guide looks at our recommended tools and conventions for testing JavaScript at Shopify. The primary focus of this guide is unit testing, but you can find some discussion of other types of testing in the [types of testing section](#types-of-testing).
+
+## Table of contents
+
+1. [Tools](#tools)
+1. [Style](#style)
+1. [Best practices](#best-practices)
+1. [Types of testing](#types-of-testing)
+1. [React](#react)
+1. [Resources](#resources)
```

Comment:
I tried to simplify the structure of the guide a bit. Only part I'm not super sure about was splitting style (stylistic rules, generally something we could lint for) and best practices (more overarching issues that affect tests pretty broadly).